amida-tech/saraswati-cql-execution
# Feature/sar 268 folder organization

# Related Tickets

- [SAR-268](https://jira.amida-tech.com/browse/SAR-168)

# How Things Worked (or Didn't) Before This PR
This is a "works better, but still room for improvement" refactor.

Before, we were processing all the measurements at once in a somewhat static way, and we were manually building value sets codes. After discussion with Mike, we'd rather each instance of `saraswati-cql-execution` execute just one measure, in a dynamic way. 

Before you ask, yes, I attempted the `cql-exec-vsac` library. It didn't seem to allow accessing any file other than the `valueset-db.json` it caches from contact with VSAC. And even when I tried that, the value sets NCQA gave us seem to want to reference their own URIs, which doesn't seem to work. If anyone sees that I did it wrong, feel free to weigh in and I'll fix it. For now, you simply point an environmental variable at value sets and it will compile the codes for you, just as `code-generator.js` does. 

# To Test

1. Choose a measure you want to cover. Put it in `/private`. For example, drop the contents of the `CISE_HEDIS_MY2022-1.0.0` folder from `ncqa-cql` into `/private`. *You no longer have to run `code-generator.js`.*
2. Update the four new environmental variables as follows:

> `MEASUREMENT_FILE`: The actual measurement file you want to run. For example, `MEASUREMENT_FILE=private\CISE_HEDIS_MY2022-1.0.0\elm\CISE_HEDIS_MY2022-1.0.0.json`
> `LIBRARIES_DIRECTORY`: The directory of the required libraries for the `MEASUREMENT_FILE`. For example, `LIBRARIES_DIRECTORY=private\CISE_HEDIS_MY2022-1.0.0\libraryElm\`
> `VALUESETS_DIRECTORY`: The directory of the required value sets for the `MEASUREMENT_FILE`. For example, `VALUESETS_DIRECTORY=private\CISE_HEDIS_MY2022-1.0.0\valuesets\`
> `MEASUREMENT_DEV_DATA`: Only for development purposes. This is the folder you want to watch with `processor.js`. For example `data\patients\immunization`.
3. Start the receiver with `node receiver.js`.
4. Run the process with `yarn localread`. Ensure you get back logging statements confirming you found the measurement file, and counts of the libraries and value sets, like so.
![image](https://user-images.githubusercontent.com/4026454/142223977-6a18ae73-de1b-40ac-80fb-c99199ced6eb.png)
5. Change something in the `MEASUREMENT_DEV_DATA` folder and confirm you see processing.
6. Do all this again, integrated in Kafka with `yarn start` to confirm that `kafka-processor.js` works. 

# Readiness



1. [ ] This PR passes all automated tests.
2. [ ] This PR has no linting errors.
3. [ ] This PR's changes to configuration files have been documented in all appropriate places (such as but not limited to README.md), if applicable.


Unit tests are coming next sprint to improve our code coverage. Stay tuned...
